recipes/bzip2/all/test_package/conanfile.py

```diff
@@ -1,17 +1,25 @@
 import os
-from conans import ConanFile, CMake, tools
+from conans import ConanFile, tools
+from conan.tools.cmake import CMake
+from conan.tools.layout import cmake_layout
 
 
 class TestPackageConan(ConanFile):
     settings = "os", "compiler", "arch", "build_type"
-    generators = "cmake", "cmake_find_package"
+    generators = "CMakeDeps", "CMakeToolchain", "VirtualRunEnv"
```

Comment:
This ``VirtualRunEnv``, we could discuss if we want it explicit or implicit. 
Its goal is to be able to run with shared libraries, the ``conan create . -o lib:shared=True`` works with this.
There is [conf] to automatically use it without explicit instantiation, but the invocation ``self.run("%s --help" % cmd, env=["conanrunenv"])`` still needs to specify ``conanrunenv``, and this is not injected by default.